### ISSUE TITLE
RET505 false positive

### ISSUE THREAD

**Issue by Kludex** [charliermarsh/ruff#2020]
# Description

I'm having a RET505 on `elif` statements, where they shouldn't be raised. 

## Code

```python
from typing import Any


def parser(obj: Any) -> str:
    if isinstance(obj, dict):
        return "a"
    elif isinstance(obj, int):
        return "b"
    return "c"
```

## Command

```bash
ruff main.py
```

## Output

```bash
main.py:5:5: RET505 Unnecessary `elif` after `return` statement
```

**Comment by bluetech** [comment]
Why shouldn't it be raised? The `elif` is indeed superfluous here (can be just `if`).

**Comment by Kludex** [comment]
Hmmm... Ok. :eyes:
